server/src/controllers/user.controllers.ts

```diff
@@ -1,100 +1,113 @@
       }
     } catch (error: any) {
-      sendResponse(res, 500, { error: error.message });
+      sendResponse(res, 500, error?.message);
     }
   } else {
-    sendResponse(res, 400, { error: 'Missing required fields' });
+    sendResponse(res, 400, 'Missing required fields');
   }
 };
 
-const getUser: RequestHandler = async (req: Request, res: Response): Promise<any> => {
+const getUser: RequestHandler = async (req: Request, res: Response): Promise<void> => {
   const { id: userId } = req.params;
 
   if (userId) {
     try {
-      const data = await User.findOne({ _id: userId }).exec();
+      const data: UserMongoType = await User.findOne({ _id: userId, isDeleted: false }).exec();
 
       if (data) {
         sendResponse(res, 200, data);
       } else {
-        sendResponse(res, 404, { error: 'User not found' });
+        sendResponse(res, 404, 'User not found');
       }
     } catch (error: any) {
-      sendResponse(res, 500, { error: error.message });
+      sendResponse(res, 500, error?.message);
     }
   } else {
     try {
-      const data = await User.find().exec();
+      const data: UserMongoType[] = await User.find({ isDeleted: false }).exec();
 
       if (data) {
         sendResponse(res, 200, data);
       } else {
-        sendResponse(res, 404, { error: 'Users not found' });
+        sendResponse(res, 404, 'Users not found');
       }
     } catch (error: any) {
-      sendResponse(res, 500, { error: error.message });
+      sendResponse(res, 500, error?.message);
     }
   }
 };
 
-const updateUser: RequestHandler = async (req: Request, res: Response): Promise<any> => {
+const updateUser: RequestHandler = async (req: Request, res: Response): Promise<void> => {
   const { id: userId, ...toUpdate } = req.body;
 
   if (userId && Object.keys(toUpdate).length > 0) {
     try {
-      const updatedUser = await User.findOneAndUpdate({ _id: userId }, toUpdate, {
-        upsert: true,
-        new: true,
-      });
+      const data: UserMongoType = await User.findOneAndUpdate(
+        { _id: userId, isDeleted: false },
+        toUpdate,
+        {
+          upsert: true,
+          new: true,
+        }
+      );
 
-      if (updatedUser) {
-        sendResponse(res, 200, updatedUser);
+      if (data) {
+        sendResponse(res, 200, data);
       } else {
-        sendResponse(res, 404, { error: 'User not found' });
+        sendResponse(res, 404, 'User not found');
       }
     } catch (error: any) {
-      sendResponse(res, 500, { error: error.message });
+      sendResponse(res, 500, error?.message);
     }
   } else {
-    sendResponse(res, 400, { error: 'Missing required fields' });
+    sendResponse(res, 400, 'Missing required fields');
   }
 };
 
-const deleteUser: RequestHandler = async (req: Request, res: Response): Promise<any> => {
-  const { id: userId } = req.params;
+const deleteUser: RequestHandler = async (req: Request, res: Response): Promise<void> => {
+  const { id: userId } = req.body;
```

Comment:
This one should be similar with the get specific id, why is this one changed to `req.body`? I think if `req.body` works too, would be better to just generalize it to `req.body` for all, so its less confusing lol